amundsen_application/static/js/components/SearchPage/tests/index.spec.tsx

```diff
@@ -155,14 +210,84 @@ describe('SearchPage', () => {
     });
 
     afterAll(() => {
-      getSelectedTabByResourceTypeSpy.mockRestore();
       createSearchOptionsSpy.mockRestore();
     });
   });
 
+  describe('componentDidUpdate', () => {
+    let props;
+    let wrapper;
+    let searchAllSpy;
+    beforeAll(() => {
+      const setupResult = setup({
+        location: {
+          search: '/search?searchTerm=current&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      });
+      props = setupResult.props;
+      wrapper = setupResult.wrapper;
+      const mockPrevProps = {
+        location: {
+          search: '/search?searchTerm=previous2&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      };
+      searchAllSpy = jest.spyOn(props, 'searchAll');
+      wrapper.instance().componentDidUpdate(mockPrevProps);
+    });
+
+    it('calls setState', () => {
+      expect(setStateSpy).toHaveBeenCalledWith({ selectedTab: ResourceType.table });
+    });
+
+    it('calls searchAll', () => {
+      expect(searchAllSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('getSanitizedUrlParams', () => {
+    let props;
+    let wrapper;
+    let getSelectedTabByResourceTypeSpy;
+
+    beforeAll(() => {
+      const setupResult = setup({
+        location: {
+          search: '/search?searchTerm=current&selectedTab=table&pageIndex=0', 
+          pathname: 'mockstr',
+          state: jest.fn(),
+          hash: 'mockstr',
+        }
+      });
+      props = setupResult.props;
+      wrapper = setupResult.wrapper;
+      const mockResourceType = ResourceType.table;
+      getSelectedTabByResourceTypeSpy = jest.spyOn(wrapper.instance(), 'getSelectedTabByResourceType').mockImplementation(() => {
+        return mockResourceType;
+      });
+      
+      wrapper.instance().getSanitizedUrlParams('current', 0, ResourceType.table)
+    });
+
+    it('calls getSelectedTabByResourceType with correct value', () => {
+      expect(getSelectedTabByResourceTypeSpy).toHaveBeenCalledWith('table');
```

Comment:
Although this works, consider that the piece of logic we're testing here at this line https://github.com/lyft/amundsenfrontendlibrary/blob/b7486b019d303fa8e3d8ef7cdcc0618b036b314b/amundsen_application/static/js/components/SearchPage/index.tsx#L123 can be summarized as "getSelectedTabByResource should be called with the 3rd parameter passed into getSantizedUrlParams". This current check doesn't truly test that previous statement, because it's leveraging our own knowledge of how the component works:
1. We know that `ResourceType.table` evaluates to the string `'table'`. We should write the tests assuming we don't know that.
2. We aren't actually testing that the value is passed appropriately, because `ResourceType.table` and the hardcoded string `'table'` are two different objects. 

What we could do instead to make this test more robust is to create an actual variable for this value and test that it is passed accordingly:
```
testTab = ResourceType.table;
wrapper.instance().getSanitizedUrlParams('current', 0, testTab);
...
expect(getSelectedTabByResourceTypeSpy).toHaveBeenCalledWith(testTab);
```